client/packages/requisitions/src/ResponseRequisition/ListView/ListView.tsx

```diff
@@ -37,33 +39,47 @@ export const ResponseRequisitionListView: FC = () => {
   } = useUrlQueryParams({ filterKey: 'comment' });
   const { data, isError, isLoading } = useResponse.document.list();
   const pagination = { page, first, offset };
+  const { authoriseCustomerRequisitions } = useResponse.utils.preferences();
   useDisableResponseRows(data?.nodes);
 
-  const columns = useColumns<ResponseRowFragment>(
+  const columnDefinitions: ColumnDescription<ResponseRowFragment>[] = [
+    [getNameAndColorColumn(), { setter: onUpdate }],
+    {
+      key: 'requisitionNumber',
+      label: 'label.number',
+      width: 100,
+    },
+    'createdDatetime',
     [
-      [getNameAndColorColumn(), { setter: onUpdate }],
+      'status',
       {
-        key: 'requisitionNumber',
-        label: 'label.number',
-        width: 100,
+        formatter: status =>
+          getRequisitionTranslator(t)(status as RequisitionNodeStatus),
       },
-      'createdDatetime',
-      [
-        'status',
-        {
-          formatter: status =>
-            getRequisitionTranslator(t)(status as RequisitionNodeStatus),
-        },
-      ],
+    ],
+  ];
+
+  if (authoriseCustomerRequisitions) {
+    columnDefinitions.push(
       // TODO Would ideally be a status (combination of requisition.status and requisition.approvalStatus) ?
-      // Should only be shown if there is store preference ? (response_requisition_requires_authorisation)
       {
         key: 'approvalStatus',
         label: 'label.auth-status',
-        width: 50,
-      },
-      ['comment', { minWidth: 400 }],
-    ],
+        minWidth: 150,
+        sortable: false,
+        accessor: ({ rowData }) =>
+          t(
+            `approval-status.${String(
+              rowData.approvalStatus
+            ).toLowerCase()}` as LocaleKey
```

Comment:
I think this makes our strong typing `weak`, if status is changed (on backend), we would have to re-run and manually catch the change, rather then having typescript show us where the problem is, also this makes it a bit harder to figure out how approval-status is mapped, personally, would prefere exhaustive matching.

Reply:
have changed to use a switch